tests/test_georaster.py

```diff
@@ -2166,44 +2182,44 @@ def test_ops_2args_implicit(self) -> None:
         # Addition
         assert r1 + r2 == self.from_array(r1.data + r2.data, rst_ref=r1)
         assert r1_f32 + r2 == self.from_array(r1_f32.data + r2.data, rst_ref=r1)
-        assert array_3d + r2 == self.from_array(array_3d + r2.data, rst_ref=r1)
-        assert r2 + array_3d == self.from_array(r2.data + array_3d, rst_ref=r1)
-        assert array_2d + r2 == self.from_array(array_2d[np.newaxis, :, :] + r2.data, rst_ref=r1)
-        assert r2 + array_2d == self.from_array(r2.data + array_2d[np.newaxis, :, :], rst_ref=r1)
-        assert r1 + floatval == self.from_array(r1.data.astype("float32") + floatval, rst_ref=r1)
```

Comment:
Ok, with reference to my comment on astype earlier, you changed the behavior.
This was by intention that, during an arithmetic operation, the dtype was changed to be compatible with the dtype of both elements. For example, if you add a float to a Raster of dtype uint8, the output was a float32. Now it will be a uint8, which means that wrapping will occur for values > 255, which can be surprising.

Reply:
The idea is simply to be fully consistent with NumPy arithmetic of MaskedArray objects. This way users can refer to NumPy directly, and we don't have to explain a special behaviour!

Surprised to hear about the issue you mention, I didn't think NumPy would not believe like (it automatically fetches the minimal common `dtype`, which would be `float` in that case).
Actually, after some tests, I cannot reproduce it:
```python
r1.data
Out[4]: 
masked_array(
  data=[[[136, 209, 209, 133, 206],
         [  0, 238, 181,  14,  84],
         [115, 109,  89,  87,  99],
         [ 28,  41, 222, 129,   6],
         [123,  91, 120, 112, 204]]],
  mask=False,
  fill_value=999999,
  dtype=uint8)
(1.5 + r1).data
Out[5]: 
masked_array(
  data=[[[137.5, 210.5, 210.5, 134.5, 207.5],
         [  1.5, 239.5, 182.5,  15.5,  85.5],
         [116.5, 110.5,  90.5,  88.5, 100.5],
         [ 29.5,  42.5, 223.5, 130.5,   7.5],
         [124.5,  92.5, 121.5, 113.5, 205.5]]],
  mask=False,
  fill_value=1e+20)
(r1 + 1.5).data
Out[6]: 
masked_array(
  data=[[[137.5, 210.5, 210.5, 134.5, 207.5],
         [  1.5, 239.5, 182.5,  15.5,  85.5],
         [116.5, 110.5,  90.5,  88.5, 100.5],
         [ 29.5,  42.5, 223.5, 130.5,   7.5],
         [124.5,  92.5, 121.5, 113.5, 205.5]]],
  mask=False,
  fill_value=1e+20)
```

We could bring this up in the meeting early October to see what's the general feeling. I'm definitely in favour of being 100% consistent with NumPy! :smile:

Reply:
But in brief, yes I agree let's stay consistent with NumPy's behavior for operations.